auto_archive.py

```diff
@@ -104,62 +105,59 @@ def process_sheet(sheet, header=1):
             archivers.WaybackArchiver(s3_client, driver)
         ]
 
-        values = gw.get_values()
         # loop through rows in worksheet
         for row in range(1 + header, gw.count_rows() + 1):
-            row_values = values[row-1]
-            url = gw.get_cell(row_values, 'url')
-            status = gw.get_cell(row_values, 'status')
+            url = gw.get_cell(row, 'url')
```

Comment:
only 1 call is made to get_values, so we can do as much get_row/get_cell as needed

Reply:
This is almost perfect, except that it was intentional that the last check of the status cell, just before starting the archive, does a new API call to the sheet. This is because there can be multiple archivers operating on a sheet simultaneously (in the case of capturing live streaming content especially).

To recap: if the "cached" status is non-blank, skip it. If the "cached" status is blank, check it again with a live API call, and then archive if it is still blank.

Make sense?

Reply:
This is almost perfect, except that it was intentional that the last check of the status cell, just before starting the archive, does a new API call to the sheet. This is because there can be multiple archivers operating on a sheet simultaneously (in the case of capturing live streaming content especially).

To recap: if the "cached" status is non-blank, skip it. If the "cached" status is blank, check it again with a live API call, and then archive if it is still blank.

Make sense?